app/permissions/PermissionsActionCheck.scala

```diff
@@ -27,6 +27,56 @@ trait PermissionActionFilter extends ActionFilter[UserRequest] with Logging {
     }
   }
 }
+trait ModifyCollectionsPermissionsCheck extends Logging { self: BaseFaciaController =>
+
+  def acl: Acl
+  def configAgent: ConfigAgent
+
+  private def testAccess(email: String, priorities: Set[PermissionsPriority], isLaunch: Boolean) = {
+
+    if (priorities.isEmpty) AccessGranted
+    else if (isLaunch)
+      acl.testUserGroupsAndCollections(
+        Permissions.LaunchEditorialFrontsAlert, Permissions.LaunchCommercialFrontsAlert,
+        Permissions.FrontsAccess, "facia-tool-allow-launch-editorial-fronts-for-all")(email, priorities)
+    else
+      acl.testUserGroupsAndCollections(
+        Permissions.EditEditorialFrontsAlert, Permissions.LaunchCommercialFrontsAlert,
+        Permissions.FrontsAccess, "facia-tool-allow-edit-editorial-fronts-for-all")(email, priorities)
+  }
+
+  def withModifyGroupPermissionForCollections[A](priorities: Set[PermissionsPriority], secondaryPriorities: Set[PermissionsPriority],
```

Comment:
Maybe define the check as an `ActionFilter` in the style of `PermissionsAccessCheck`? That would avoid an extra level of nesting in the calling code.

I think we'd need to pass in `configAgent` to the trait, maybe as a def that's then overridden in each controller.

Reply:
This is difficult to do because we don't actually know which permissions to check when we hit the endpoint so I think we should leave it as it as discussed.